provisioning/transport/mqtt/test/_mqtt_test.js

```diff
@@ -250,6 +255,32 @@ describe('Mqtt', function () {
 
     });
 
+    ['$rid=123&retry-after=456&something=89','retry-after=456&$rid=123&something=89','retry-after=456&something=89&$rid=123'].forEach(function(theQueryString) {
+      it ('can parse response with multiple query parameters', function(callback) {
+        var fakeUuids = [uuid.v4()];
+        var uuidStub = sinon.stub(uuid,'v4');
+        uuidStub.returns(fakeUuids[0]);
+
+        mqtt.queryOperationStatus(fakeRequest, fakeOperationId, function(err, result) {
+          uuid.v4.restore();
+          assert.oneOf(err, [null, undefined]);
+          assert.deepEqual(result, fakeResponse);
+          callback();
+        });
+        extendedResponse(theQueryString.replace('$rid=123', '$rid='+fakeUuids[0]));
+      });
+    });
+
+    ['$rid=123',''].forEach(function(theQueryString) {
```

Comment:
>['$rid=123',''] [](start = 4, length = 15)

pls add a comment to explain why these aren't proper topics

Reply:
done.

---
In reply to: [280189337](https://github.com/Azure/azure-iot-sdk-node/pull/549#discussion_r280189337) [](ancestors = 280189337)